### PR TITLE
fix(plugins-documentation): restrict access handle ctx.session undefined

### PR DESCRIPTION
**Pull request by sjoukedv**
### What does it do?

For restricting the access to the documentation page in the plugin; use `?.` notation to handle `ctx.session` undefined.

### Why is it needed?

```shell
[2023-03-13 10:02:03.477] error: Cannot read properties of undefined (reading 'documentation')
TypeError: Cannot read properties of undefined (reading 'documentation')
    at module.exports (/repo/dir/node_modules/@strapi/plugin-documentation/server/middlewares/restrict-access.js:12:20)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async policiesMiddleware (/repo/dir/node_modules/@strapi/strapi/lib/services/server/policy.js:24:5)
    at async serve (/repo/dir/node_modules/koa-static/index.js:59:5)
    at async returnBodyMiddleware (/repo/dir/node_modules/@strapi/strapi/lib/services/server/compose-endpoint.js:52:18)
    at async policiesMiddleware (/repo/dir/node_modules/@strapi/strapi/lib/services/server/policy.js:24:5)
    at async /repo/dir/dist/src/index.js:224:17
    at async /repo/dir/node_modules/@strapi/strapi/lib/middlewares/body.js:58:9
    at async /repo/dir/node_modules/@strapi/strapi/lib/middlewares/logger.js:25:5
    at async /repo/dir/node_modules/@strapi/strapi/lib/middlewares/powered-by.js:16:5
```

### How to test it?

Access the `/documentation` path for a restricted page without being logged in.

### Related issue(s)/PR(s)

Not found

### CODE REVIEW

**Comment by joshuaellis** [comment]
Hey, can you please elaborate on `Why is it needed?` i.e. this error doesn't tell me much e.g "how did you get this error?" because it may be that while this fixes the surface issue or stopping the error, there's a bigger issue i.e. it might be the case it should always be defined and it not being so means theres a bigger problem that should be fixed instead :)

**Comment by sjoukedv** [comment]
> Hey, can you please elaborate on `Why is it needed?` i.e. this error doesn't tell me much e.g "how did you get this error?" because it may be that while this fixes the surface issue or stopping the error, there's a bigger issue i.e. it might be the case it should always be defined and it not being so means theres a bigger problem that should be fixed instead :)

@joshuaellis it throws 500 server errors on the path that holds the swagger UI:

```json
{"data":null,"error":{"status":500,"name":"InternalServerError","message":"Internal Server Error"}}
```

Because `ctx.session` is undefined adding the `?.` notations prevents the code from throwing an error. If `ctx.session` is `undefined` this part of the statement yields `false`, but the first condition is `ctx.session.documentation` assumes session exists and it can't access the `documentation` field of `undefined`. Instead of `?.` we can also check first for `!ctx.session` and the lazy evaluation / "short-circuit" evaluation will stop the evaluation of the remaining conditions.

After this, sending

```
GET /documentation HTTP/1.1
Accept-Encoding: gzip, deflate, br
Accept-Language: nl,en-US;q=0.9,en;q=0.8
Cache-Control: no-cache
Connection: keep-alive
Host: localhost:1337
...
```

returns the expected redirect to the login page

```
HTTP/1.1 302 Found
...
Location: http://localhost:1337/documentation/login
Content-Type: text/html; charset=utf-8
Connection: keep-alive
Keep-Alive: timeout=5
```

File: packages/plugins/documentation/server/middlewares/restrict-access.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ module.exports = async (ctx, next) => {
     return next();
   }
 
-  if (!ctx.session.documentation || !ctx.session.documentation.logged) {
+  if (!ctx.session?.documentation || !ctx.session?.documentation.logged) {
```
Comment (remidej) [inline packages/plugins/documentation/server/middlewares/restrict-access.js:12]: With optional chaining you can safely combine them:

```suggestion
  if (!ctx.session?.documentation?.logged) {
```

**Review by remidej — COMMENTED** [review]
Hello and sorry for not looking at this earlier. I'm not able to reproduce the issue you want to fix. Could you check if it still occurs on the latest strapi version?

**Pull request closed without merging**